engine/platform/psvita/sys_psvita.c

```diff
@@ -75,32 +75,63 @@ const size_t __vrtld_num_exports = sizeof( aux_exports ) / sizeof( *aux_exports
 
 /* end of export crap */
 
+static const char *PSVita_GetLaunchParameter( char *outbuf )
+{
+	SceAppUtilAppEventParam param;
+	memset( &param, 0, sizeof( param ) );
+	sceAppUtilReceiveAppEvent( &param );
+	if( param.type == 0x05 )
+	{
+		sceAppUtilAppEventParseLiveArea( &param, outbuf );
+		return outbuf;
+	}
+	return NULL;
+}
+
 void Platform_ShellExecute( const char *path, const char *parms )
 {
 	Con_Reportf( S_WARN "Tried to shell execute ;%s; -- not supported\n", path );
 }
 
 void PSVita_Init( void )
 {
+	string errorMsg;
+	char param[2048] = { 0 };
 	char xashdir[1024] = { 0 };
+	SceAppUtilInitParam initParam = { 0 };
+	SceAppUtilBootParam bootParam = { 0 };
 
 	// cd to the base dir immediately for library loading to work
-	if ( PSVita_GetBasePath( xashdir, sizeof( xashdir ) ) )
+	if( PSVita_GetBasePath( xashdir, sizeof( xashdir )))
 	{
 		chdir( xashdir );
 	}
 
-	sceCtrlSetSamplingModeExt( SCE_CTRL_MODE_ANALOG_WIDE );
-	sceTouchSetSamplingState( SCE_TOUCH_PORT_FRONT, SCE_TOUCH_SAMPLING_STATE_START );
+	sceAppUtilInit( &initParam, &bootParam );
+
+	// check if the user wants debug mode
+	if( PSVita_GetLaunchParameter( param ))
+	{
+		if( !Q_strcmp( param, "dev" ))
+		{
+			Con_Reportf( S_NOTE "Enabled developer mode from launcher.\n" );
```

Comment:
What if we append cmdline instead, append `-dev 2 -log` there and let the engine automatically enable developer mode with enabled logging without platform-specific behavior in `engine/common`?

Reply:
Well, `-dev` checked before `Platform_Init` is called.

What if you reimplmement `main()` from `common/launcher.c` to make it PSVita-specific that reads livearea parameters and sets the appropriate argc and argv?

Reply:
You can take out main() function from `common/launcher.c` (this file REALLY needs to be decoupled to platform-specific launchers, or removed completely, but we'll do that later =/), remove anything that's not relevant to `psvita` and insert your livearea check there.

Reply:
In relation to the previous review and this one, like you said, you can't trivially inject arguments in `Platform_Init` and I wanted to keep all Vita-related initialization in there. But if that's not a requirement then sure.

Reply:
Yeah, but we really need to override `main()` for platforms. I think it's a good compromise between hacking the engine functions, and have two stage initialization for platforms.